reference/forms/types/choice.rst

```diff
@@ -184,6 +184,19 @@ The ``choice_loader`` can be used to only partially load the choices in cases wh
 a fully-loaded list is not necessary. This is only needed in advanced cases and
 would replace the ``choices`` option.
 
+You can use an instance of :class:`Symfony\\Component\\Form\\ChoiceList\\Loader\\CallbackChoiceLoader` 
+if you want to take advantage of lazy loading::
+
+    use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+    use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+    // ...
+
+    $builder->add('constants', ChoiceType::class, array(
+        'choice_loader' => new CallbackChoiceLoader(function() {
+            return StaticClass::getConstants();
+        },
+    ));
+
```

Comment:
We could precise that, the call won't happen if there is a redirection (i.e the view is not built), AND if there is no pre set data or submitted data. Because otherwise we'll need to transform the value to a model choice (or the opposite) and that implies loading choices too.